packages/cactus-plugin-cc-tx-visualization/src/main/typescript/plugin-cc-tx-visualization.ts

```diff
@@ -0,0 +1,453 @@
+    });
+    this.prometheusExporter =
+      options.prometheusExporter ||
+    new PrometheusExporter({ pollingIntervalInMin: 1 });
+    this.instanceId = this.options.instanceId;
+    this.crossChainLog = new CrossChainEventLog({name:"CC-TX-VIZ_EVENT_LOGS"});
+    //todo should allow different models to be instantiated
+    this.crossChainModel = new CrossChainModel();
+    this.txReceipts = [];
+    this.persistMessages = options.channelOptions.persistMessages || false;
+    this.eventProvider = options.eventProvider;
+    this.log.debug("Initializing connection to RabbitMQ");
+    this.amqpConnection = new Amqp.Connection(this.eventProvider);
+    this.log.info("Connection to RabbitMQ server initialized");
+    this.amqpExchange = this.amqpConnection.declareExchange(`cc-tx-viz-exchange`, "direct", {durable: this.persistMessages});
+    this.amqpQueue = this.amqpConnection.declareQueue(this.queueId, {durable: this.persistMessages});
+    this.amqpQueue.bind(this.amqpExchange);
+  
+    const finalTime = new Date();
+    this.log.debug(`EVAL-${this.className}-SETUP-CONSTRUCTOR:${finalTime.getTime()-startTime.getTime()}`);
+
+  } 
+  getOpenApiSpec(): unknown {
+    throw new Error("Method not implemented.");
+  }
+
+  get numberEventsLog(): number {
+    return this.crossChainLog.numberEvents();
+  }
+
+  get numberUnprocessedReceipts(): number {
+    return this.txReceipts.length;
+  }  
+
+  public purgeCrossChainEvents(): void {
+    this.crossChainLog.purgeLogs();
+  }
+
+  // todo connection closing is  problematic, tests are left hanging
+  public async closeConnection(): Promise<void>  {
+    this.log.debug("Closing Amqp connection");
+    try {
+      this.amqpConnection.close();
+      this.log.debug(" Amqp connection closed");
+
+    } catch (error) {
+      this.log.error(error);
+    }
+
+  }
+
+  public getInstanceId(): string {
+    return this.instanceId;
+  }
+
+  public async onPluginInit(): Promise<unknown> {
+    return;
+  }
+
+  public getPrometheusExporter(): PrometheusExporter {
+    return this.prometheusExporter;
+  }
+
+  public async getPrometheusExporterMetrics(): Promise<string> {
+    const res: string = await this.prometheusExporter.getPrometheusMetrics();
+    this.log.debug(`getPrometheusExporterMetrics() response: %o`, res);
+    return res;
+  }
+
+  public async shutdown(): Promise<void> {
+    this.log.info(`Shutting down...`);
+    const serverMaybe = this.getHttpServer();
+    if (serverMaybe.isPresent()) {
+      this.log.info(`Awaiting server.close() ...`);
+      const server = serverMaybe.get();
+      await promisify(server.close.bind(server))();
+      this.log.info(`server.close() OK`);
+    } else {
+      this.log.info(`No HTTP server found, skipping...`);
+    }
+  }
+
+  public async registerWebServices(
+    app: Express,
+  ): Promise<IWebServiceEndpoint[]> {
+    const webApp: Express = this.options.webAppOptions ? express() : app;
+
+    if (this.options.webAppOptions) {
```

Comment:
@RafaelAPB The plugins hosting their own web servers has been deprecated a while back IIRC. Could you please migrate to defining an endpoint class and then registering that with the API server (by returning it through `getOrCreateWebServices()`)

Reply:
Done